extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/job/CompositeJob.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.astro.internal.job;
+
+import java.util.List;
+
+/**
+ * {@link CompositeJob} comprises multiple {@link Job}s to be executed in order
+ *
+ * @author Markus Rathgeb - Initial contribution
+ * @author Amit Kumar Mondal - Minor modifications
+ */
+public final class CompositeJob extends AbstractJob {
+
+    private final String thingUID;
+    private final List<Job> jobs;
+
+    /**
+     * Constructor
+     * 
+     * @param jobs the jobs to execute
+     * @throws IllegalArgumentException
+     *             if {@code jobs} is {@code null} or empty
+     */
+    public CompositeJob(List<Job> jobs) {
+        checkArgument(jobs != null, "Jobs must not be null");
+        checkArgument(!jobs.isEmpty(), "Jobs must not be empty");
+
+        this.jobs = jobs;
+        this.thingUID = jobs.get(0).getThingUID();
+
+        boolean notMatched = jobs.stream().anyMatch(j -> !j.getThingUID().equals(thingUID));
+        checkArgument(!notMatched, "The jobs must associate the same thing UID");
+    }
+
+    @Override
+    public String getThingUID() {
+        return thingUID;
+    }
+
+    @Override
+    public void run() {
+        jobs.forEach(Job::run);
```

Comment:
Iterating over the list using the `Iterable#forEach` is inherently thread-safe while iterating through a collection. I opted `Stream#forEach` out of this because it will not give us ordered execution. This is my primary motive. I could add logging behavior to catch the Runtime Exception as well.